src/index.ts

```diff
@@ -0,0 +1,77 @@
+import { PinejsClientCoreFactory } from 'pinejs-client-core';
+export type { PinejsClientCoreFactory } from 'pinejs-client-core';
+
+interface BackendParams {
+	/** The browser fetch API implementation or a compatible one */
+	fetch?: typeof fetch;
+}
+
+type PromiseObj = Promise<{}>;
+
+class RequestError extends Error {
+	public code = 'PineClientFetchRequestError';
+
+	constructor(
+		public body: string,
+		public status: number,
+		public responseHeaders: object,
+	) {
+		super(`Request error: ${body}`);
+	}
+}
+
+export default class PineFetch extends PinejsClientCoreFactory(Promise)<
+	PineFetch,
+	PromiseObj,
+	Promise<PinejsClientCoreFactory.PromiseResultTypes>
+> {
+	constructor(
+		params: PinejsClientCoreFactory.Params,
+		public backendParams: BackendParams,
+	) {
+		super(params);
+		if (!backendParams?.fetch && typeof fetch === 'undefined') {
```

Comment:
It'd be nice to also add a check for `typeof backendParams.fetch === 'function'`